x-pack/platform/plugins/shared/fleet/public/applications/integrations/sections/epm/screens/installed_integrations/components/installed_integrations_table.tsx

```diff
@@ -279,13 +286,18 @@ export const InstalledIntegrationsTable: React.FunctionComponent<{
                   ),
                   icon: 'trash',
                   type: 'icon',
-                  description: i18n.translate(
-                    'xpack.fleet.epmInstalledIntegrations.uninstallIntegrationLabel',
-                    {
-                      defaultMessage: 'Uninstall integration',
-                    }
-                  ),
+
                   onClick: (item) => bulkUninstallIntegrationsWithConfirmModal([item]),
+                  enabled: (item) => !usedByAgentPolicy(item),
+                  description: (item) =>
+                    i18n.translate(
+                      'xpack.fleet.epmInstalledIntegrations.uninstallIntegrationLabel',
+                      {
+                        defaultMessage: usedByAgentPolicy(item)
+                          ? 'You cannot uninstall this integration because it is used by one or more agent policies'
```

Comment:
Same as above.

```suggestion
                          ? "You can't uninstall this integration because it is used by one or more agent policies"
```